xbmc/cores/AudioEngine/Utils/PackerMAT.h

```diff
@@ -0,0 +1,106 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <deque>
+#include <stdint.h>
+#include <vector>
+
+struct TrueHDMajorSyncInfo
+{
+  int ratebits{0};
+  uint16_t outputTiming{0};
+  bool outputTimingPresent{false};
+  bool valid{false};
+};
+
+enum class Type : bool
+{
+  PADDING = true,
+  DATA = false,
+};
+
+class CPackerMAT
+{
+public:
+  CPackerMAT();
+  ~CPackerMAT() = default;
+
+  bool PackTrueHD(const uint8_t* data, int size);
+  bool HaveOutput() const { return !m_outputQueue.empty(); }
+  std::vector<uint8_t> GetOutputFrame();
+
+private:
+  struct MATState
+  {
+    bool init;
+    int ratebits;
+    uint16_t outputTiming;
+    bool outputTimingValid;
+
+    uint16_t prevFrametime;
+    bool prevFrametimeValid;
+
+    uint32_t matFramesize;
+    uint32_t prevMatFramesize;
+
+    uint32_t padding;
+    uint32_t nSamples;
+    int nSamplesOffset;
```

Comment:
samplesOffset

Reply:
i think the N holds useful explination here, rather than saying this is an offset from "samples" (whatever that is), its saying the number of samples offset. Which I dont think is implied from the variable samplesOffset

Reply:
I think clarity through language is better than through convention. So if it's desired to have it define the amount I would rather use numberOfSamplesOffset or samplesOffsetCount.

Reply:
Added code comments to clarify the meaning of each variable

Reply:
> I think clarity through language is better than through convention. So if it's desired to have it define the amount I would rather use numberOfSamplesOffset or samplesOffsetCount.

I agree on this point, the n made sense to me as I understand the context, to others without it later could infer other meaning. We're on the same page! :)